beacon_node/store/src/metadata.rs

```diff
@@ -95,12 +95,19 @@ pub struct AnchorInfo {
     pub state_upper_limit: Slot,
     /// The slot before which historical states are available (<=).
     pub state_lower_limit: Slot,
+    /// This represents the earliest slot that we download historical blocks to when doing a
+    /// weak-subjectivity sync.
+    /// This can be the genesis slot, if configured to download blocks all the way back to
+    /// genesis.
+    // NOTE: Although this is always an epoch boundary, we use the Slot type here to avoid
+    // extra generic type parameters.
+    pub weak_subjectivity_point: Slot,
```

Comment:
If we add to the `AnchorInfo` we need a database schema migration, as it is stored on disk.

Maybe we could avoid that by storing the `weak_subjectivity_point` in memory and passing it to `block_backfill_complete`?

Reply:
oh didn't realise it required a db scheme shift :scream: 

Have you got suggestions for where in memory would be a better place for this?

I could thread it into network sync land, but I avoided it because the anchor seemed to be the cleanest way. Going into sync land requires a bit of parameter passing.

Reply:
We could slap it on the `beacon_chain`, or keep it in the anchor and turn off SSZ encoding/decoding for it (which will prevent it being stored). I think I prefer the `beacon_chain` approach. We could initialize it on start-up using the clock-based logic you've already implemented.

Reply:
We should also change the condition here so that state reconstruction only runs if we've backfilled all the way to 0:

https://github.com/sigp/lighthouse/blob/38514c07f222ff7783834c48cf5c0a6ee7f346d0/beacon_node/beacon_chain/src/historical_blocks.rs#L196-L200

Might need a new method like `anchor.block_backfill_complete_to_genesis()`.